source/shared_lib/meson.build

```diff
@@ -0,0 +1,168 @@
+# subdir('sources/streflop')
+# subdir('sources/util')
+subdir('sources/feathery_ftp')
+
+src = []
+
+src_basename = [
+  'string_utils.cpp',
+  'util.cpp',
+  'properties.cpp',
+  'profiler.cpp',
+  'leak_dumper.cpp',
+  'checksum.cpp',
+  'conversion.cpp',
+  'randomgen.cpp',
+]
+
+foreach file : src_basename
+    src += [join_paths('sources/util/', file)]
+endforeach
+
+src_basename = [
+  'base_thread.cpp',
+  'byte_order.cpp',
+  'cache_manager.cpp',
+  'platform_common.cpp',
+  'simple_threads.cpp',
+]
+
+foreach file : src_basename
+    src += [join_paths('sources/platform/common/', file)]
+endforeach
+
+src_basename = [
+  'ircclient.cpp',
+  'miniftpclient.cpp',
+  'miniftpserver.cpp',
+  'socket.cpp',
+]
+
+foreach file : src_basename
+    src += [join_paths('sources/platform/posix/', file)]
+endforeach
+
+src_basename = [
+  'factory_repository.cpp',
+  'gl_wrap.cpp',
+  'platform_util.cpp',
+  'thread.cpp',
+  'window.cpp',
+  'window_gl.cpp',
+]
+
+foreach file : src_basename
+    src += [join_paths('sources/platform/sdl/', file)]
+endforeach
+
+dep_curl = dependency('libcurl')
+dep_sdl = dependency('sdl2')
+dep_x11 = dependency('x11')
+dep_openal = dependency('openal')
+
+# Using simply 'lua' doesn't work on all Linux distributions
+# See https://github.com/mesonbuild/meson/issues/4553 for more info
+foreach name : ['lua', 'lua5.3', 'lua-5.3', 'lua53']
+    dep_lua = dependency(name, version: '>=5.3', required: false)
+    if dep_lua.found()
+        break
+    endif
+endforeach
+if not dep_lua.found()
+    error('Lua could not be found!')
+endif
+
+dep_cppunit = dependency('cppunit')
+dep_fribidi = dependency('fribidi')
+dep_glib = dependency('glib-2.0')
+
+inc_ircclient = []
```

Comment:
I asked about (reported) this in https://github.com/mesonbuild/meson/issues/10738

Too bad I forgot to specifically ask if there was a better way to write this condition for libircclient. Maybe I'll do that later.